flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/StreamTableEnvironment.scala

```diff
@@ -92,8 +93,7 @@ abstract class StreamTableEnvironment(
   def ingest(tableName: String): Table = {
 
     if (isRegistered(tableName)) {
-      relBuilder.scan(tableName)
-      new Table(relBuilder.build(), this)
+      new Table(this, CatalogNode(tableName, getTable(tableName), getTypeFactory))
     }
     else {
       throw new TableException(s"Table \'$tableName\' was not found in the registry.")
```

Comment:
I think a ValidationException would be more appropriate, because `ingest()` belongs to a query IMO.